doc/services/crypto/psa_crypto.rst

```diff
@@ -0,0 +1,195 @@
+.. _psa_crypto:
+
+PSA Crypto
+##########
+
+Overview
+********
+
+The PSA Crypto API offers a portable programming interface for
+cryptographic operations and key storage across a wide range of
+hardware. It is designed to be user-friendly while still providing
+access to the low-level primitives essential for modern cryptography.
+
+It is created and maintained by Arm. Arm developed the PSA (Platform
+Security Architecture) as a comprehensive security framework to
+address the increasing security needs of connected devices.
+
+In Zephyr, PSA Crypto API is implemented using mbedTLS, an
+open-source cryptographic library that provides the underlying
+cryptographic functions.
+
+Design Goals
+************
+
+The interface is suitable for a vast range of devices: from
+special-purpose cryptographic processors that process data with a
+built-in key, to constrained devices running custom application code,
+such as microcontrollers, and multi-application devices, such as
+servers. It follows the principle of cryptographic agility.
+
+* Algorithm Flexibility
+
+  PSA Crypto API supports a wide range of cryptographic algorithms,
```

Comment:
please consider using the following construct ("[definition lists](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#lists-and-quote-like-blocks)")  for all these types of enumerations in the file:

```suggestion
Algorithm Flexibility
  PSA Crypto API supports a wide range of cryptographic algorithms,
```

This makes things much easier to read

![image](https://github.com/zephyrproject-rtos/zephyr/assets/128251/346cbc88-fb91-4339-b314-eb3ec37e1973)

Reply:
Looks good to me. Do we have a design guideline for documentation ? If don't would be nice to have these guidelines documented to get a consistent look'n feel